project-addons/flask_middleware_connector/models/stock.py

```diff
@@ -29,13 +32,24 @@ class StockMove(models.Model):
     _inherit = "stock.move"
 
     #TODO: Debería ser al asignar un producto, al cancelarlo, al finalizarlo y al eliminar la reserve
-
     @api.multi
     def write(self, vals):
         res = super(StockMove, self).write(vals)
-        if vals.get('state', False) and vals["state"] != "draft":
-            for move in self:
-                session = ConnectorSession(self.env.cr, self.env.uid,
+        for move in self:
+            if vals.get('picking_id', False) or (vals.get('state', False) and move.picking_id):
+                vals_picking = {'state': move.picking_id.state}
+                if not vals_picking['state']:
```

Comment:
Si state no viniera en vals falla. Sería: if not vals.get('state')

Reply:
En el primer if se controla si viene state en vals. Si solo viene picking_id en vals, metemos el estado de este en el diccionario, ya que despues del super().write() el picking ya tiene el estado final.